sky/sdk/lib/widgets/variable_height_scrollable.dart

```diff
@@ -14,18 +12,28 @@ class VariableHeightScrollable extends Scrollable {
   VariableHeightScrollable({
     String key,
     this.builder,
-    this.token
+    this.token,
+    this.layoutState
   }) : super(key: key);
 
   IndexedBuilder builder;
   Object token;
+  BlockViewportLayoutState layoutState;
   bool _contentsChanged = true;
 
+  void initState() {
+    if (layoutState == null)
```

Comment:
This seems just like a convenience: if layoutState is passed in then we use it, otherwise we create one.